Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -91,154 +87,66 @@ void RawDataReaderCheck::configure()
-          h->Scale(1. / h->GetBinContent(mIndexMBclass));
-        }
-      }
-
-      if (cycleCounter > 1) { // skipping the check for first two cycles, as the first one is arbitrary long
-        TH1F* fHistDifference = (TH1F*)h->Clone();
-        TH1F* fHistPrev;
-        if (inputRates && !relativeRates) {
-          if (fHistInputPrevious) {
-            fHistPrev = (TH1F*)fHistInputPrevious->Clone();
-          } else {
-            fHistInputPrevious = (TH1F*)h->Clone();
-            continue;
-          }
-        }
-        if (inputRates && relativeRates) {
-          if (fHistInputRatioPrevious) {
-            fHistPrev = (TH1F*)fHistInputRatioPrevious->Clone();
-          } else {
-            fHistInputRatioPrevious = (TH1F*)h->Clone();
-            continue;
-          }
-        }
-        if (!inputRates && !relativeRates) {
-          if (fHistClassesPrevious) {
-            fHistPrev = (TH1F*)fHistClassesPrevious->Clone();
-          } else {
-            fHistClassesPrevious = (TH1F*)h->Clone();
-            continue;
-          }
-        }
-        if (!inputRates && relativeRates) {
-          if (fHistClassRatioPrevious) {
-            fHistPrev = (TH1F*)fHistClassRatioPrevious->Clone();
-          } else {
-            fHistClassRatioPrevious = (TH1F*)h->Clone();
-            continue;
-          }
-        }
-        if (!relativeRates) {
-          fHistDifference->Scale(1. / (mCycleDuration * mFraction));
-          fHistPrev->Scale(1. / (mCycleDuration * mFraction));
-        }
-        fHistDifference->Add(fHistPrev, -1); // Calculate relative difference w.r.t. rate in previous cycle
-        fHistDifference->Divide(fHistPrev);
-
-        float thrBad = mThresholdRateBad;
-        float thrMedium = mThresholdRateMedium;
-        if (relativeRates) {
-          thrBad = mThresholdRateRatioBad;
-          thrMedium = mThresholdRateRatioMedium;
-        }
-        for (size_t i = 1; i < fHistDifference->GetXaxis()->GetNbins() + 1; i++) { // Check how many inputs/classes changed more than a threshold value
-          if (TMath::Abs(fHistDifference->GetBinContent(i)) > thrBad) {
-            vIndexBad.push_back(i);
-          } else if (TMath::Abs(fHistDifference->GetBinContent(i)) > thrMedium) {
-            vIndexMedium.push_back(i);
-          }
-        }
-        delete fHistDifference;
-        delete fHistPrev;
-      }
-      if (inputRates && !relativeRates) {
+    } else if (mo->getName() == "inputs") {
+      if (fHistInputPrevious) {
+        checkChange(h, fHistInputPrevious, vIndexBad, vIndexMedium);
         delete fHistInputPrevious;
-        fHistInputPrevious = (TH1F*)h->Clone();
       }
-      if (inputRates && relativeRates) {
+      fHistInputPrevious = (TH1F*)h->Clone();
+    } else if (mo->getName() == "inputRatio") {
+      if (fHistInputRatioPrevious) {
         delete fHistInputRatioPrevious;
-        fHistInputRatioPrevious = (TH1F*)h->Clone();
       }
```

Comment:
@lietava It seems to me that `inputRatio` is not checked, and also the re-scaling with respec to the MB trigger (if I understand correctly the purpose of the ratio plots) is not applied in the task. Can you check?